Sources/MapboxMaps/Location/Puck/Puck2D.swift

```diff
@@ -45,10 +65,12 @@ internal final class Puck2D: NSObject, Puck {
 
     internal init(configuration: Puck2DConfiguration,
                   style: StyleProtocol,
-                  locationProducer: LocationProducerProtocol) {
+                  interpolatedLocationProducer: InterpolatedLocationProducerProtocol) {
         self.configuration = configuration
         self.style = style
-        self.locationProducer = locationProducer
+        self.interpolatedLocationProducer = interpolatedLocationProducer
+        // swiftlint:disable:next force_cast
+        self.encodedScale = (try! JSONSerialization.jsonObject(with: JSONEncoder().encode([configuration.resolvedScale])) as! [Any]).first!
```

Comment:
scale is a Value which can be either a constant or an expression. This converts the value to a json object that we can pass in the properties dictionary. I'll try to make it a bit more safe…

Reply:
Actually, this isn't any less safe than before. Previously, resolvedScale was set as the value for LocationIndicatorLayer's topImageSize, bearingImageSize, and shadowImageSize properties and then we were converting the LocationIndicatorLayer to a JSON object using `try! layer.jsonObject()`.

The only way this line would trap is if resolvedScale throws an error when it gets encoded. If the encoding succeeds, we can trust that decoding to a JSON object will also succeed unless JSONEncoder itself produces invalid JSON (we trust that this won't happen), and if decoding to a JSON object succeeds, then we know that the force cast to `[Any]` will succeed since we hard-code the input to be an array, and if the force cast succeeds, we know that the array will contain a single element (again since we hard-code that in the input), so the force-unwrap of `.first` will always succeed.

The reason that the input is wrapped in an array is that encoding and decoding JSON fragments aren't supported on some older versions of iOS that we still support.

That said, maybe I can wrap this up into a nicer abstraction :)

Reply:
Thank you for the detailed explanation! Initially, I failed to notice that `resolvedScale` is a `Value<Double>`, not just a `Double` type. Very good reasoning about JSON encoding and decoding as well. I love how it's nicely tucked away into an extension now 😍